determined-ai/determined
# chore: properly handle external logout

## Description

When using "external sessions" configuration in managed clusters, we are running into scenarios where the UI gets stuck after token expiration. This PR updates the master API to return 401 on token expiration and takes a user back through the login flow if external sessions are enabled and a 401 is returned. 


## Test Plan

Manually tested the web UI against:

Regular Determined / MLDE cluster run via devcluster
Cluster configured with external sessions

Steps:

- Validate there is no change to logout functionality if external sessions are not enabled. 

- If external sessions are enabled, validate that a user is logged back in on token expiration and logout works as expected.



## Checklist

- User-facing API changes need the "User-facing API Change" label.
- Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- Licenses should be included for new code which was copied and/or modified from any external code.
